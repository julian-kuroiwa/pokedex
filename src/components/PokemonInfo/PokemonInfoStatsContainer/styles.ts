import styled from 'styled-components';

export const Container = styled.div`
  max-width: 400px;
  margin: 40px auto;
  width: 100%;
`;
