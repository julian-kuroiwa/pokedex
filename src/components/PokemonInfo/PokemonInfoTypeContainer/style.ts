import styled from 'styled-components';

export const Container = styled.ul`
  list-style: none;
  width: 100%;
  display: flex;
  margin-top: 10px;
`;
