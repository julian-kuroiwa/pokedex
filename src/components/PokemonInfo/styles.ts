import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  max-width: 600px;
  margin: 40px auto;

  header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--primary-color);
    padding: 20px;
    color: #fff;
    font-size: 18px;

    > span {
      font-weight: bold;
    }
  }

  ul {
    list-style: none;
    width: 100%;
    display: flex;
    margin: 20px 0;

    > li {
      padding: 5px 10px;
      background-color: yellow;
      color: var(--text-color);

      & + li {
        margin-left: 10px;
      }
    }
  }

  img {
    width: 200px;
  }
`;
