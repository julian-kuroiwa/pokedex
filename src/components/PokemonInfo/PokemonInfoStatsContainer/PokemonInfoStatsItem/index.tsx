import React from 'react';

import { Container } from './styles';

interface StatProps {
  stat: {
    base_stat: number;
    stat: {
      name: string;
    };
  };
}

const PokemonInfoStatsItem: React.FC<StatProps> = ({ stat }) => (
  <Container>
    <strong>{stat.stat.name}:</strong>
    <span>{stat.base_stat}</span>
  </Container>
);

export default PokemonInfoStatsItem;
