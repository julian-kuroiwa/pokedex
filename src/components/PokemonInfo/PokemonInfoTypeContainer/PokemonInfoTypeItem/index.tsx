import React from 'react';

import { Container } from './style';

interface TypeProps {
  type: string;
}

const PokemonInfoTypeItem: React.FC<TypeProps> = ({ type }) => (
  <Container typeName={type}>{type}</Container>
);

export default PokemonInfoTypeItem;
