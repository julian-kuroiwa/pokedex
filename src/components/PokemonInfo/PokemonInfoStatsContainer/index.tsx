import React from 'react';

import { Container } from './styles';
import PokemonInfoStatsItem, { PokemonStats } from './PokemonInfoStatsItem';

interface PokemonStatsProps {
  pokemonStats: PokemonStats[];
}

const PokemonInfoStatsContainer: React.FC<PokemonStatsProps> = ({
  pokemonStats,
}) => (
  <Container>
    {pokemonStats.map(stat => (
      <PokemonInfoStatsItem key={stat.stat.name} stat={stat} />
    ))}
  </Container>
);

export default PokemonInfoStatsContainer;
