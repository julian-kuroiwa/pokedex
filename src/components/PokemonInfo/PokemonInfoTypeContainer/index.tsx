import React from 'react';

import { Container } from './style';

import PokemonInfoTypeItem from './PokemonInfoTypeItem';

export interface PokemonTypes {
  type: {
    name: string;
  };
}

interface PokemonTypesProps {
  pokemonTypes: PokemonTypes[];
}

const PokemonInfoTypeContainer: React.FC<PokemonTypesProps> = ({
  pokemonTypes,
}) => (
  <Container>
    {pokemonTypes.map(type => (
      <PokemonInfoTypeItem key={type.type.name} type={type.type.name}>
        {type.type.name}
      </PokemonInfoTypeItem>
    ))}
  </Container>
);

export default PokemonInfoTypeContainer;
