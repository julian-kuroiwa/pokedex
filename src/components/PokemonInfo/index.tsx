import React from 'react';

import { Container } from './styles';

import PokemonInfoTypeContainer, {
  PokemonTypes,
} from './PokemonInfoTypeContainer';
import PokemonInfoStatsContainer from './PokemonInfoStatsContainer';
import { PokemonStats } from './PokemonInfoStatsContainer/PokemonInfoStatsItem';

export interface PokemonProps {
  id: number;
  name: string;
  sprites: {
    front_default: string;
  };
  stats: PokemonStats[];
  types: PokemonTypes[];
}

interface PokemonInfoProps {
  pokemon: PokemonProps;
}

const PokemonInfo: React.FC<PokemonInfoProps> = ({ pokemon }) => {
  return (
    <Container>
      <header>
        <h2>{pokemon.name}</h2>
        <span>#{pokemon.id}</span>
      </header>

      <div>
        <PokemonInfoTypeContainer pokemonTypes={pokemon.types} />

        <img src={pokemon.sprites.front_default} alt={pokemon.name} />

        <PokemonInfoStatsContainer pokemonStats={pokemon.stats} />
      </div>
    </Container>
  );
};

export default PokemonInfo;
