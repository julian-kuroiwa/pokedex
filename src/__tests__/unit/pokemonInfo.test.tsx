import { mount } from 'enzyme';
import React from 'react';
import PokemonInfo from '../../components/PokemonInfo';
import PokemonInfoTypeContainer from '../../components/PokemonInfo/PokemonInfoTypeContainer';
import PokemonInfoStatsContainer from '../../components/PokemonInfo/PokemonInfoStatsContainer';

describe('PokemonInfo', () => {
  let props;

  beforeEach(() => {
    props = {
      pokemon: {
        name: 'ditto',
        id: 132,
        types: [
          {
            type: {
              name: 'normal',
            },
          },
        ],
        sprites: {
          front_default:
            'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png',
        },
        stats: [
          {
            base_stat: '48',
            stat: {
              name: 'speed',
            },
          },
          {
            base_stat: '48',
            stat: {
              name: 'special-defense',
            },
          },
          {
            base_stat: '48',
            stat: {
              name: 'special-attack',
            },
          },
          {
            base_stat: '48',
            stat: {
              name: 'defense',
            },
          },
          {
            base_stat: '48',
            stat: {
              name: 'attack',
            },
          },
          {
            base_stat: '48',
            stat: {
              name: 'hp',
            },
          },
        ],
      },
    };
  });
  it('should match snapshot', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the name and id correctly', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    expect(wrapper.find('h2').text()).toBe(props.pokemon.name);
    expect(wrapper.find('span').text()).toBe(`#${props.pokemon.id}`);
  });

  it('should render PokemonInfoTypeContainer', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    expect(wrapper.find(PokemonInfoTypeContainer)).toBeTruthy();
  });

  it('should render each type properly', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    wrapper.find(PokemonInfoTypeContainer).forEach((node, index) => {
      expect(node.text()).toContain(props.pokemon.types[index].type.name);
    });
  });

  it('should render PokemonInfoStatsContainer', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    expect(wrapper.find(PokemonInfoStatsContainer)).toBeTruthy();
  });

  it('should render each stat properly', () => {
    const wrapper = mount(<PokemonInfo {...props} />);

    wrapper.find(PokemonInfoStatsContainer).forEach((node, index) => {
      expect(node.text()).toContain(props.pokemon.stats[index].stat.name);
    });
  });
});
