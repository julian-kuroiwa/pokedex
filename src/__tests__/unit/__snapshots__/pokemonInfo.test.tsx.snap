// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonInfo should match snapshot 1`] = `
<PokemonInfo
  pokemon={
    Object {
      "id": 132,
      "name": "ditto",
      "sprites": Object {
        "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png",
      },
      "stats": Array [
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "speed",
          },
        },
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "special-defense",
          },
        },
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "special-attack",
          },
        },
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "defense",
          },
        },
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "attack",
          },
        },
        Object {
          "base_stat": "48",
          "stat": Object {
            "name": "hp",
          },
        },
      ],
      "types": Array [
        Object {
          "type": Object {
            "name": "normal",
          },
        },
      ],
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm hbUrdP"
    >
      <header>
        <h2>
          ditto
        </h2>
        <span>
          #
          132
        </span>
      </header>
      <div>
        <PokemonInfoTypeContainer
          pokemonTypes={
            Array [
              Object {
                "type": Object {
                  "name": "normal",
                },
              },
            ]
          }
        >
          <styled.ul>
            <ul
              className="sc-AxirZ eNJIJQ"
            >
              <PokemonInfoTypeItem
                key="normal"
                type="normal"
              >
                <styled.li
                  typeName="normal"
                >
                  <li
                    className="sc-AxiKw iaYPAM"
                  >
                    normal
                  </li>
                </styled.li>
              </PokemonInfoTypeItem>
            </ul>
          </styled.ul>
        </PokemonInfoTypeContainer>
        <img
          alt="ditto"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
        />
        <PokemonInfoStatsContainer
          pokemonStats={
            Array [
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "speed",
                },
              },
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "special-defense",
                },
              },
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "special-attack",
                },
              },
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "defense",
                },
              },
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "attack",
                },
              },
              Object {
                "base_stat": "48",
                "stat": Object {
                  "name": "hp",
                },
              },
            ]
          }
        >
          <styled.div>
            <div
              className="sc-AxhCb iyIKL"
            >
              <PokemonInfoStatsItem
                key="speed"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "speed",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      speed
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
              <PokemonInfoStatsItem
                key="special-defense"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "special-defense",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      special-defense
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
              <PokemonInfoStatsItem
                key="special-attack"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "special-attack",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      special-attack
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
              <PokemonInfoStatsItem
                key="defense"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "defense",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      defense
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
              <PokemonInfoStatsItem
                key="attack"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "attack",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      attack
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
              <PokemonInfoStatsItem
                key="hp"
                stat={
                  Object {
                    "base_stat": "48",
                    "stat": Object {
                      "name": "hp",
                    },
                  }
                }
              >
                <styled.li
                  baseStat="48"
                >
                  <li
                    className="sc-AxhUy cZeKdq"
                  >
                    <strong>
                      hp
                      :
                    </strong>
                    <div />
                  </li>
                </styled.li>
              </PokemonInfoStatsItem>
            </div>
          </styled.div>
        </PokemonInfoStatsContainer>
      </div>
    </div>
  </styled.div>
</PokemonInfo>
`;
